source/high-order-euler.cc

```diff
@@ -0,0 +1,698 @@
+#include <deal.II/numerics/vector_tools.h>
+#include <deal.II/base/parameter_acceptor.h>
+
+//xbraid include
+#include <braid.h>
+#include <braid_test.h>
+
+//mgrit includes
+#include "level_structures.h"
+
+/*
+ * Author: Jerett Cherry, Colorado State University
+ */
+
+const int dim = 2;
+/**
+ *   \brief Contains the implementation of the mandatory X-Braid functions
+ *
+ *  X-Braid mandates several functions in order to drive the solution.
+ *  This file contains the implementation of said mandatory functions.
+ *  See the X-Braid documentation for more information.
+ *  There are several functions that are optional in X-Braid that may
+ *  or may not be implemented in here.
+ */
+
+/*Includes*/
+
+/*-------- Third Party --------*/
+
+/*-------- Project --------*/
+//#include "offlinedata.h"
+//#include "parallel-in-time-euler.h"
+
+// This preprocessor macro is used on function arguments
+// that are not used in the function. It is used to
+// suppress compiler warnings.
+#define UNUSED(x) (void)(x)
+
+
+/**
+ * This function should take a vector and resize it with
+ * the size specified
+ */
+
+template<typename Description, typename V, int dim>
+void resizeVector(V& v,
+                  const int size,
+                  const int num_comps = ryujin::TimeLoop<Description,
+                                                         dim,
+                                                         NUMBER>::problem_dimension)
+{
+  for(int i = 0; i < num_comps; ++i)
+    v[i].reinit(size);
+}
+
+
+// This struct contains all data that changes with time. For now
+// this is just the solution data. When doing AMR this should
+// probably include the triangulization, the sparsity patter,
+// constraints, etc.
+/**
+ * \brief Struct that contains the deal.ii vector of size (dim+2,n_dofs).
+ */
+typedef struct _braid_Vector_struct
+{
+  ryujin::TimeLoop<ryujin::Euler::Description, dim, NUMBER>::vector_type U;//change this to
+
+} my_Vector;
+
+
+// This struct contains all the data that is unchanging with time.
+/**
+ *
+ */
+typedef struct _braid_App_struct : public dealii::ParameterAcceptor
+{
+    using Description = ryujin::Euler::Description;
+    using Number = NUMBER;
+    using LevelType = std::shared_ptr<ryujin::mgrit::LevelStructures<Description, 2, Number>>;
+
+  public:
+
+    using HyperbolicSystemView =
+        typename Description::HyperbolicSystem::template View<dim, Number>;
+
+    static constexpr unsigned int problem_dimension =
+        HyperbolicSystemView::problem_dimension;
+    static constexpr unsigned int n_precomputed_values =
+        HyperbolicSystemView::n_precomputed_values;
+    using scalar_type = typename ryujin::OfflineData<dim, Number>::scalar_type;
+    using vector_type = ryujin::MultiComponentVector<Number, problem_dimension>;
+    using precomputed_type = ryujin::MultiComponentVector<Number, n_precomputed_values>;
+
+
+    MPI_Comm comm_x, comm_t;//fixme:should these be const somehow?
```

Comment:
```suggestion
    const MPI_Comm comm_x, comm_t;
```
You are not modifying those, so make them const.